sdk/core/azure-core/src/main/java/com/azure/core/credential/package-info.java

```diff
@@ -2,6 +2,90 @@
 // Licensed under the MIT License.
 
 /**
- * Package containing basic credential classes for authentication purposes.
+ * <p>Azure Core Credential library is designed to simplify the process of authenticating and authorizing access
+ * to Azure services from Java applications. The SDK provides a set of classes and methods that handle authentication
+ * and credential management, allowing developers to securely connect to Azure services without dealing with the
+ * low-level details of authentication protocols.</p>
+ *
+ * <p>The library provides a unified way to obtain credentials for various Azure authentication
+ * mechanisms, such as Azure Active Directory (AAD), shared access signatures and API keys. It abstracts the
+ * complexities of authentication and provides a consistent programming model for accessing Azure services.</p>
+ *
+ * <p>By using the library, users can easily integrate Azure authentication into their applications, retrieve the
+ * required credentials based on the desired authentication method, and use those credentials to authenticate
+ * requests to Azure services like Azure Storage, Azure Key Vault, Azure Service Bus, and more.</p>
+ *
+ * <p>The library offers several authentication types for authenticating with Azure services. Here are some of the
+ * authentication mechanisms supported by the library:</p>
+ * <ol>
+ *     <li>Azure Active Directory (AAD) Authentication.</li>
+ *     <li>Shared Access Signature (SAS) Authentication</li>
+ *     <li>Key Based Authentication.</li>
```

Comment:
Why do some of these list elements end in `.` and others don't? Since these are short statements I'd have none end in `.`